### DISCUSSION TITLE
Command option to go offline without running anything

### DISCUSSION THREAD

**Discussion by brunoborges** [jbangdev/jbang#1108] [Q&A]
As a user, I'd like to create a Dockerfile and during build time, cache all required deps (JDK and libraries), and also prep the app for faster startup when the container starts.

Suggestion:

```
$ jbang prepare <app>
```

As an example:

```Dockerfile
FROM jbangdev/jbang-action

RUN echo "eula=true" > eula.txt

ENTRYPOINT ["jbang", "minecraft-server@microsoft"]
```

With this, the JDK and the JAR file will be downloaded at every new container.

**Comment by maxandersen** [comment]
two options:

 `RUN jbang build minecraft-server@microsoft` and it will build so its cached.
or if you want to aviod any overhead do:

```
RUN jbang export minecraft-server@microsoft
ENTRYPOINT ["jbang", "minecraft-server.jar"]
```

**Reply by quintesse** [reply]
Of course the **real** way to make a tiny build that only builds once is this: https://github.com/quintesse/door-opener-proxy/blob/main/Dockerfile 😉
